File: program/program/implement-selection-sort/implement-selection-sort.swift
func selectionSort<T: Comparable>(_ array: inout [T]) {
    guard array.count > 1 else {
        return
    }

    for i in 0..<array.count - 1 {
        var minIndex = i

        for j in i + 1..<array.count {
            if array[j] < array[minIndex] {
                minIndex = j
            }
        }

        if i != minIndex {
            array.swapAt(i, minIndex)
        }
    }
}

// Example usage:
var numbers = [64, 34, 25, 12, 22, 11, 90]
selectionSort(&numbers)
print("Sorted array: \(numbers)")

File: program/implement-selection-sort/ImplementSelectionSort.js
function selectionSort(arr) {
    const n = arr.length;

    for (let i = 0; i < n - 1; i++) {
        // Assume the current index is the minimum
        let minIndex = i;

        // Find the index of the minimum element in the remaining unsorted array
        for (let j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }

        // Swap the found minimum element with the element at the current index
        if (minIndex !== i) {
            [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
        }
    }
}

// Example usage:
let arrayToSort = [64, 25, 12, 22, 11];
selectionSort(arrayToSort);
console.log("Sorted array:", arrayToSort);
